lib/label_templates/code39_vertical_tube.rb

```diff
@@ -0,0 +1,42 @@
+require './lib/label_template_setup'
+
+def tube_barcode_definition(name, type_id, barcode_type)
+  {
+    name: name,
+    label_type_id: type_id,
+    labels_attributes:[{
+      name: 'label',
+      bitmaps_attributes:[
+        {"horizontal_magnification":"05","vertical_magnification":"05","font":"H","space_adjustment":"03","rotational_angles":"11",
+      "x_origin":"0042","y_origin":"0210","field_name":"bottom_line"},
+    {"horizontal_magnification":"05","vertical_magnification":"05","font":"H","space_adjustment":"02","rotational_angles":"11",
+      "x_origin":"0042","y_origin":"0010","field_name":"middle_line"},
```

Comment:
Just checking this "y_origin":"0010" is intended to change from "0210" to "0010"? (Because the bottom_line and top_line ones are still "0210")

Reply:
In this template field_name doesn't represent where the text is really located in the label. Originally field names were named so we could have a description on where in the label they are going to be printed.
The fields have been rearranged in this new label distribution, but I could not change the field name without making more changes. I'll add it as a separate work to do.